cleanlab_studio/internal/tlm/validation.py

```diff
@@ -0,0 +1,178 @@
+import os
+from typing import Union, Sequence, Any
+from cleanlab_studio.errors import ValidationError
+from cleanlab_studio.internal.constants import (
+    _VALID_TLM_MODELS,
+    TLM_MAX_TOKEN_RANGE,
+    TLM_NUM_CANDIDATE_RESPONSES_RANGE,
+    TLM_NUM_CONSISTENCY_SAMPLES_RANGE,
+)
+
+
+SKIP_VALIDATE_TLM_OPTIONS = os.environ.get("CLEANLAB_STUDIO_SKIP_VALIDATE_TLM_OPTIONS", False)
+
+
+def validate_tlm_prompt(prompt: Union[str, Sequence[str]]) -> None:
+    if isinstance(prompt, str):
+        return
+
+    elif isinstance(prompt, Sequence):
+        if any(not isinstance(p, str) for p in prompt):
+            raise ValidationError(
+                "Some items in prompt are of invalid types, all items in the prompt list must be of type str."
+            )
+
+    else:
+        raise ValidationError(
+            f"Invalid type {type(prompt)}, prompt must either be strings or list/iterable of strings."
+        )
+
+
+def validate_tlm_try_prompt(prompt: Sequence[str]) -> None:
```

Comment:
not a blocker for this PR, but there is a decent amount of code duplication in this file. if we plan on adding more types of queries than prompt and response in the future, then it'll be easier to maintain with a helper function that checks type(s) of string/sequence as string.